src/bodywork/git.py

```diff
@@ -44,102 +50,162 @@ def download_project_code_from_repo(
     """
     log = bodywork_log_factory()
     try:
-        run(['git', '--version'], check=True, stdout=DEVNULL)
+        run(["git", "--version"], check=True, stdout=DEVNULL)
     except CalledProcessError:
-        raise RuntimeError('git is not available')
+        raise RuntimeError("git is not available")
     try:
-        if (get_connection_protocol(url) is ConnectionPrototcol.SSH
-                and get_remote_repo_host(url) is GitRepoHost.GITHUB):
-            setup_ssh_for_github()
-        elif SSH_GITHUB_KEY_ENV_VAR not in os.environ:
-            log.warning('Not configured for use with private GitHub repos')
+        if get_connection_protocol(url) is ConnectionProtocol.SSH:
+            hostname = urlparse(f"ssh://{url}").hostname
+            if hostname is not None and hostname != "":
+                setup_ssh_for_git_host(hostname)
+            else:
+                raise ValueError(
+                    f"Unable to derive hostname from URL {url}. Please check "
+                )
+        elif SSH_PRIVATE_KEY_ENV_VAR not in os.environ:
+            log.warning("Not configured for use with private GitHub repos")
     except Exception as e:
-        msg = f'Unable to setup SSH for Github and you are trying to connect via SSH: {e}' # noqa
+        msg = f"Unable to setup SSH for Github and you are trying to connect via SSH: {e}"  # noqa
```

Comment:
Probably don't need the `# noqa` after it has been Blacked?

Reply:
👍